packages/slate-editor/src/modules/nodes-hierarchy/normilizers/removeWhenNoChildren.ts

```diff
@@ -0,0 +1,16 @@
+import { Node, Transforms } from 'slate';
+
+import type { HierarchyNormalizer } from '../types';
+
+export function removeWhenNoChildren(): HierarchyNormalizer {
+    return (editor, path) => {
+        const node = Node.get(editor, path);
+
+        if ('children' in node && node.children.length === 0) {
+            Transforms.removeNodes(editor, { at: path, match: (n) => n === node });
+            return true;
+        }
+
+        return false;
+    };
+}
```

Comment:
Nitpicking.

Do you notice how `mustHaveChildren()` and `removeWhenNoChildren()` are almost the same?

I think they can be one *rule* with different fixes:

```ts
requireAtLeastOneChild(insertChild(createParagraph))
requireAtLeastOneChild(removeNode)
```

Reply:
Yes, i thought about that, however this condition `requireAtLeastOneChild(removeNode)` makes me confused as it tells that it requires a child inside and at the same time removes whole node
I was thinking about conditions in naming
```
whenNoChild(insertChild(createParagraph))
whenNoChild(removeNode)
```
but as i remember we are not going this way

Reply:
> makes me confused as it tells that it requires a child inside and at the same time removes whole node

IDK, I see no problem. :man_shrugging: